pkg/util/strategicpatch/patch.go

```diff
@@ -167,6 +213,12 @@ func mergeSlice(original, patch []interface{}, elemType reflect.Type, mergeKey s
 	if t.Kind() != reflect.Map {
 		// Maybe in the future add a "concat" mode that doesn't
 		// uniqify.
+		if !reflect.DeepEqual(original, patch) {
+			dw.WriteRemoval(fmt.Sprintf("%v", original))
+			dw.WriteAddition(fmt.Sprintf("%v", patch))
+		} else {
+			dw.WriteSame(fmt.Sprintf("%v", original))
+		}
```

Comment:
@caesarxuchao I'm not certain. Your comment makes sense, but I'm having trouble putting it into the context of a diff. I'm kinda working without a spec concerning how the diff should appear. From your example, a resultant diff would be
```
  a
- b
- c
+ d
```
Correct?